[bug] The SCons generator doesn't export system_libs

<!--
  Please don't forget to update the issue title.
  Include all applicable information to help us reproduce your problem.

  To help us debug your issue please explain:
-->

As a bit of background, I recently had a very strange issue with icu/67.1. Specifically, when I built the release build, it produced linking errors. The debug build was strangely fine, but this turned out to be random.

A few rabbit holes and a few hours later, I managed to figure out why -- I wasn't linking against `-ldl` and `-lpthread`. Linking against only `-ldl` didn't cause link errors, but it caused `System error: -1` (or something along the lines of that) at runtime.

I decided to run the linker into verbose mode, and noticed several libraries were missing. I cross-referenced with the recipe, and in the  debug build, the flags weren't present. I checked the  SConscript file generated several times, but the last time around I noticed that there was no -pthread or -ldl. I also just confirmed these two were only included because of my use of sanitizers - it working this far has been pure luck.

Finally, the problem itself, and the reason I'm opening this issue here, is that the SCons generator [doesn't export system_libs](https://github.com/conan-io/conan/blob/develop/conans/client/generators/scons.py#L15). That's why it fails to link or run.

### Environment Details (include every applicable attribute)
  * Operating System+version: Linux Mint 19.3
  * Compiler+version: Clang 10
  * Conan version: 1.27.0
  * Python version: 3.8.0

### Steps to reproduce (Include if Applicable)

conanfile.py:
```python
from conans import ConanFile

class Taskranger(ConanFile):
    settings = "os", "compiler", "build_type", "arch"
    generators = "ycm", "scons"

    default_options = {
        "icu:data_packaging": "static",
        "icu:shared": False
    }

    def requirements(self):
        self.requires("icu/67.1")
```

main.cpp:
```cpp
#include "unicode/smpdtfmt.h"

int main() {
    using namespace icu;
    UErrorCode status = U_ZERO_ERROR;
    SimpleDateFormat sdf(UnicodeString("dd.MM.y"), status);
}
```
SConstruct:
```python
env = Environment(CXXFLAGS="-std=c++17")
flags = SConscript("build/SConscript_conan")
env.MergeFlags(flags["conan"])
print(flags)
env.Program("demo", "main.cpp")
```
Commands after placing the files above in a directory:
```
mkdir build
cd build
conan install .. --build missing
scons ..
```
... and watch the compiler come up with undefined references to `-ldl` stuff. With `with_dyload: False`, it will compile, but it throws an `std::system_error` because pthreads isn't linked. [Both of these are defined as `system_libs`](https://github.com/conan-io/conan-center-index/blob/master/recipes/icu/all/conanfile.py#L233-L235), neither are included in the flags generated by the SCons generator. This should in theory, in a simple enough environment, happen to any and all packages using system_libs.

Output with:

Default settings:
```
g++ -o main.o -c -std=c++17 -DU_STATIC_IMPLEMENTATION -I/home/lunarwatcher/.conan/data/icu/67.1/_/_/package/b840526c6db636afcf60cda7b7728f7e08d6bb6e/include main.cpp
g++ -o demo main.o -L/home/lunarwatcher/.conan/data/icu/67.1/_/_/package/b840526c6db636afcf60cda7b7728f7e08d6bb6e/lib -licui18n -licuio -licutest -licutu -licuuc -licudata
/home/lunarwatcher/.conan/data/icu/67.1/_/_/package/b840526c6db636afcf60cda7b7728f7e08d6bb6e/lib/libicuuc.a(putil.ao): In function `uprv_dl_open_67':
putil.cpp:(.text.uprv_dl_open_67+0x13): undefined reference to `dlopen'
/home/lunarwatcher/.conan/data/icu/67.1/_/_/package/b840526c6db636afcf60cda7b7728f7e08d6bb6e/lib/libicuuc.a(putil.ao): In function `uprv_dl_close_67':
putil.cpp:(.text.uprv_dl_close_67+0x7): undefined reference to `dlclose'
/home/lunarwatcher/.conan/data/icu/67.1/_/_/package/b840526c6db636afcf60cda7b7728f7e08d6bb6e/lib/libicuuc.a(putil.ao): In function `uprv_dlsym_func_67':
putil.cpp:(.text.uprv_dlsym_func_67+0xe): undefined reference to `dlsym'
collect2: error: ld returned 1 exit status
scons: *** [demo] Error 1
scons: building terminated because of errors.
```

`with_dyload: False`:
```
❯ ./demo
terminate called after throwing an instance of 'std::system_error'
  what():  Unknown error -1
[1]    6878 abort (core dumped)  ./demo
```

--- a/conans/client/generators/scons.py
+++ b/conans/client/generators/scons.py
@@ -1,3 +1,7 @@
+import textwrap
+
+from jinja2 import Template
+
 from conans.model import Generator
 
 
@@ -8,41 +12,47 @@ class SConsGenerator(Generator):
 
     @property
     def content(self):
-        template = ('    "{dep}" : {{\n'
-                    '        "CPPPATH"     : {info.include_paths},\n'
-                    '        "LIBPATH"     : {info.lib_paths},\n'
-                    '        "BINPATH"     : {info.bin_paths},\n'
-                    '        "LIBS"        : {libs},\n'
-                    '        "FRAMEWORKS"  : {info.frameworks},\n'
-                    '        "FRAMEWORKPATH"  : {info.framework_paths},\n'
-                    '        "CPPDEFINES"  : {info.defines},\n'
-                    '        "CXXFLAGS"    : {info.cxxflags},\n'
-                    '        "CCFLAGS"     : {info.cflags},\n'
-                    '        "SHLINKFLAGS" : {info.sharedlinkflags},\n'
-                    '        "LINKFLAGS"   : {info.exelinkflags},\n'
-                    '    }},\n'
-                    '    "{dep}_version" : "{info.version}",\n')
+        template = textwrap.dedent("""\
+            "{{dep}}" : {
+                "CPPPATH"     : {{info.include_paths}},
+                "LIBPATH"     : {{info.lib_paths}},
+                "BINPATH"     : {{info.bin_paths}},
+                {% set libs = info.libs + info.system_libs -%}
+                "LIBS"        : [{% if libs %}'{{ libs|join("', '") }}'{% endif %}],
+                "FRAMEWORKS"  : {{info.frameworks}},
+                "FRAMEWORKPATH"  : {{info.framework_paths}},
+                "CPPDEFINES"  : {{info.defines}},
+                "CXXFLAGS"    : {{info.cxxflags}},
+                "CCFLAGS"     : {{info.cflags}},
+                "SHLINKFLAGS" : {{info.sharedlinkflags}},
+                "LINKFLAGS"   : {{info.exelinkflags}},
+            },
+            "{{dep}}_version" : "{{info.version}}",
+            """)
+        # Nice nested indent
+        template = "\n".join("    " + line for line in template.splitlines())
+        template = Template(template)
 
         sections = ["conan = {\n"]
 
         libs = self.deps_build_info.libs + self.deps_build_info.system_libs
-        all_flags = template.format(dep="conan", info=self.deps_build_info, libs=libs)
+        all_flags = template.render(dep="conan", info=self.deps_build_info, libs=libs)
         sections.append(all_flags)
 
         for config, cpp_info in self.deps_build_info.configs.items():
             libs = cpp_info.libs + cpp_info.system_libs
-            all_flags = template.format(dep="conan:" + config, info=cpp_info, libs=libs)
+            all_flags = template.render(dep="conan:" + config, info=cpp_info, libs=libs)
             sections.append(all_flags)
 
         for dep_name, info in self.deps_build_info.dependencies:
             dep_name = dep_name.replace("-", "_")
             libs = info.libs + info.system_libs
-            dep_flags = template.format(dep=dep_name, info=info, libs=libs)
+            dep_flags = template.render(dep=dep_name, info=info, libs=libs)
             sections.append(dep_flags)
 
             for config, cpp_info in info.configs.items():
                 libs = cpp_info.libs + cpp_info.system_libs
-                all_flags = template.format(dep=dep_name + ":" + config, info=cpp_info, libs=libs)
+                all_flags = template.render(dep=dep_name + ":" + config, info=cpp_info, libs=libs)
                 sections.append(all_flags)
 
         sections.append("}\n")

--- a/conans/client/generators/scons.py
+++ b/conans/client/generators/scons.py
@@ -13,6 +13,8 @@ class SConsGenerator(Generator):
                     '        "LIBPATH"     : {info.lib_paths},\n'
                     '        "BINPATH"     : {info.bin_paths},\n'
                     '        "LIBS"        : {libs},\n'
+                    '        "FRAMEWORKS"  : {info.frameworks},\n'
+                    '        "FRAMEWORKPATH"  : {info.framework_paths},\n'
                     '        "CPPDEFINES"  : {info.defines},\n'
                     '        "CXXFLAGS"    : {info.cxxflags},\n'
                     '        "CCFLAGS"     : {info.cflags},\n'

--- a/conans/client/generators/scons.py
+++ b/conans/client/generators/scons.py
@@ -12,7 +12,7 @@ class SConsGenerator(Generator):
                     '        "CPPPATH"     : {info.include_paths},\n'
                     '        "LIBPATH"     : {info.lib_paths},\n'
                     '        "BINPATH"     : {info.bin_paths},\n'
-                    '        "LIBS"        : {info.libs},\n'
+                    '        "LIBS"        : {libs},\n'
                     '        "CPPDEFINES"  : {info.defines},\n'
                     '        "CXXFLAGS"    : {info.cxxflags},\n'
                     '        "CCFLAGS"     : {info.cflags},\n'
@@ -21,23 +21,26 @@ class SConsGenerator(Generator):
                     '    }},\n'
                     '    "{dep}_version" : "{info.version}",\n')
 
-        sections = []
-        sections.append("conan = {\n")
+        sections = ["conan = {\n"]
 
-        all_flags = template.format(dep="conan", info=self.deps_build_info)
+        libs = self.deps_build_info.libs + self.deps_build_info.system_libs
+        all_flags = template.format(dep="conan", info=self.deps_build_info, libs=libs)
         sections.append(all_flags)
 
         for config, cpp_info in self.deps_build_info.configs.items():
-            all_flags = template.format(dep="conan:" + config, info=cpp_info)
+            libs = cpp_info.libs + cpp_info.system_libs
+            all_flags = template.format(dep="conan:" + config, info=cpp_info, libs=libs)
             sections.append(all_flags)
 
         for dep_name, info in self.deps_build_info.dependencies:
             dep_name = dep_name.replace("-", "_")
-            dep_flags = template.format(dep=dep_name, info=info)
+            libs = info.libs + info.system_libs
+            dep_flags = template.format(dep=dep_name, info=info, libs=libs)
             sections.append(dep_flags)
 
             for config, cpp_info in info.configs.items():
-                all_flags = template.format(dep=dep_name + ":" + config, info=cpp_info)
+                libs = cpp_info.libs + cpp_info.system_libs
+                all_flags = template.format(dep=dep_name + ":" + config, info=cpp_info, libs=libs)
                 sections.append(all_flags)
 
         sections.append("}\n")

--- a/conans/client/generators/scons.py
+++ b/conans/client/generators/scons.py
@@ -35,24 +35,20 @@ class SConsGenerator(Generator):
 
         sections = ["conan = {\n"]
 
-        libs = self.deps_build_info.libs + self.deps_build_info.system_libs
-        all_flags = template.render(dep="conan", info=self.deps_build_info, libs=libs)
+        all_flags = template.render(dep="conan", info=self.deps_build_info)
         sections.append(all_flags)
 
         for config, cpp_info in self.deps_build_info.configs.items():
-            libs = cpp_info.libs + cpp_info.system_libs
-            all_flags = template.render(dep="conan:" + config, info=cpp_info, libs=libs)
+            all_flags = template.render(dep="conan:" + config, info=cpp_info)
             sections.append(all_flags)
 
         for dep_name, info in self.deps_build_info.dependencies:
             dep_name = dep_name.replace("-", "_")
-            libs = info.libs + info.system_libs
-            dep_flags = template.render(dep=dep_name, info=info, libs=libs)
+            dep_flags = template.render(dep=dep_name, info=info)
             sections.append(dep_flags)
 
             for config, cpp_info in info.configs.items():
-                libs = cpp_info.libs + cpp_info.system_libs
-                all_flags = template.render(dep=dep_name + ":" + config, info=cpp_info, libs=libs)
+                all_flags = template.render(dep=dep_name + ":" + config, info=cpp_info)
                 sections.append(all_flags)
 
         sections.append("}\n")

--- a/conans/test/unittests/client/generators/scons_test.py
+++ b/conans/test/unittests/client/generators/scons_test.py
@@ -1,7 +1,8 @@
+import platform
 import unittest
 
 from conans.client.generators.scons import SConsGenerator
-from conans.model.build_info import CppInfo
+from conans.model.build_info import CppInfo, DepCppInfo
 from conans.model.conan_file import ConanFile
 from conans.model.env_info import EnvValues
 from conans.model.ref import ConanFileReference
@@ -23,13 +24,13 @@ class SConsGeneratorTest(unittest.TestCase):
         cpp_info = CppInfo(ref.name, "")
         cpp_info.defines = ["MYDEFINE2"]
         cpp_info.version = "3.2.3"
-        conanfile.deps_cpp_info.add(ref.name, cpp_info)
+        conanfile.deps_cpp_info.add(ref.name, DepCppInfo(cpp_info))
         generator = SConsGenerator(conanfile)
         content = generator.content
         scons_lines = content.splitlines()
-        self.assertIn("        \"CPPDEFINES\"  : [\'MYDEFINE2\', \'MYDEFINE1\'],", scons_lines)
-        self.assertIn("        \"CPPDEFINES\"  : [\'MYDEFINE1\'],", scons_lines)
-        self.assertIn("        \"CPPDEFINES\"  : [\'MYDEFINE2\'],", scons_lines)
+        self.assertIn("        \"CPPDEFINES\"  : ['MYDEFINE2', 'MYDEFINE1'],", scons_lines)
+        self.assertIn("        \"CPPDEFINES\"  : ['MYDEFINE1'],", scons_lines)
+        self.assertIn("        \"CPPDEFINES\"  : ['MYDEFINE2'],", scons_lines)
         self.assertIn('    "conan_version" : "None",', scons_lines)
         self.assertIn('    "MyPkg_version" : "0.1",', scons_lines)
         self.assertIn('    "MyPkg2_version" : "3.2.3",', scons_lines)
@@ -48,7 +49,11 @@ class SConsGeneratorTest(unittest.TestCase):
         conanfile.deps_cpp_info.add("MyPkg", DepCppInfo(cpp_info))
         generator = SConsGenerator(conanfile)
         content = generator.content
-        self.assertIn('"LIBS"        : [\'mypkg\', \'pthread\']', content)
-        self.assertIn('"FRAMEWORKS"  : [\'cocoa\']', content)
-        self.assertIn('"FRAMEWORKPATH"  : [\'/rootpath/frameworks\']', content)
-        self.assertIn('"MyPkg_version" : "0.1"', content)
+        scons_lines = content.splitlines()
+        self.assertIn('        "LIBS"        : [\'mypkg\', \'pthread\'],', scons_lines)
+        self.assertIn('        "FRAMEWORKS"  : [\'cocoa\'],', scons_lines)
+        if platform.system() == "Windows":
+            self.assertIn('        "FRAMEWORKPATH"  : [\'/rootpath\\\\frameworks\'],', scons_lines)
+        else:
+            self.assertIn('        "FRAMEWORKPATH"  : [\'/rootpath/frameworks\'],', scons_lines)
+        self.assertIn('    "MyPkg_version" : "0.1",', scons_lines)

--- a/conans/test/unittests/client/generators/scons_test.py
+++ b/conans/test/unittests/client/generators/scons_test.py
@@ -33,3 +33,17 @@ class SConsGeneratorTest(unittest.TestCase):
         self.assertIn('    "conan_version" : "None",', scons_lines)
         self.assertIn('    "MyPkg_version" : "0.1",', scons_lines)
         self.assertIn('    "MyPkg2_version" : "3.2.3",', scons_lines)
+
+    def system_libs_test(self):
+        # https://github.com/conan-io/conan/issues/7301
+        conanfile = ConanFile(TestBufferConanOutput(), None)
+        conanfile.initialize(Settings({}), EnvValues())
+        cpp_info = CppInfo("MyPkg", "")
+        cpp_info.version = "0.1"
+        cpp_info.libs = ["mypkg"]
+        cpp_info.system_libs = ["pthread"]
+        conanfile.deps_cpp_info.add("MyPkg", cpp_info)
+        generator = SConsGenerator(conanfile)
+        content = generator.content
+        self.assertIn('"LIBS"        : [\'mypkg\', \'pthread\']', content)
+        self.assertIn('"MyPkg_version" : "0.1"', content)

--- a/conans/test/unittests/client/generators/scons_test.py
+++ b/conans/test/unittests/client/generators/scons_test.py
@@ -38,7 +38,7 @@ class SConsGeneratorTest(unittest.TestCase):
         # https://github.com/conan-io/conan/issues/7301
         conanfile = ConanFile(TestBufferConanOutput(), None)
         conanfile.initialize(Settings({}), EnvValues())
-        cpp_info = CppInfo("MyPkg", "")
+        cpp_info = CppInfo("MyPkg", "/rootpath")
         cpp_info.version = "0.1"
         cpp_info.libs = ["mypkg"]
         cpp_info.system_libs = ["pthread"]

--- a/conans/test/unittests/client/generators/scons_test.py
+++ b/conans/test/unittests/client/generators/scons_test.py
@@ -34,7 +34,7 @@ class SConsGeneratorTest(unittest.TestCase):
         self.assertIn('    "MyPkg_version" : "0.1",', scons_lines)
         self.assertIn('    "MyPkg2_version" : "3.2.3",', scons_lines)
 
-    def system_libs_test(self):
+    def system_frameworks_libs_test(self):
         # https://github.com/conan-io/conan/issues/7301
         conanfile = ConanFile(TestBufferConanOutput(), None)
         conanfile.initialize(Settings({}), EnvValues())
@@ -42,8 +42,13 @@ class SConsGeneratorTest(unittest.TestCase):
         cpp_info.version = "0.1"
         cpp_info.libs = ["mypkg"]
         cpp_info.system_libs = ["pthread"]
+        cpp_info.frameworks = ["cocoa"]
+        cpp_info.frameworkdirs = ["frameworks"]
+        cpp_info.filter_empty = False
         conanfile.deps_cpp_info.add("MyPkg", cpp_info)
         generator = SConsGenerator(conanfile)
         content = generator.content
         self.assertIn('"LIBS"        : [\'mypkg\', \'pthread\']', content)
+        self.assertIn('"FRAMEWORKS"  : [\'cocoa\']', content)
+        self.assertIn('"FRAMEWORKPATH"  : [\'frameworks\']', content)
         self.assertIn('"MyPkg_version" : "0.1"', content)

--- a/conans/test/unittests/client/generators/scons_test.py
+++ b/conans/test/unittests/client/generators/scons_test.py
@@ -50,5 +50,5 @@ class SConsGeneratorTest(unittest.TestCase):
         content = generator.content
         self.assertIn('"LIBS"        : [\'mypkg\', \'pthread\']', content)
         self.assertIn('"FRAMEWORKS"  : [\'cocoa\']', content)
-        self.assertIn('"FRAMEWORKPATH"  : [\'frameworks\']', content)
+        self.assertIn('"FRAMEWORKPATH"  : [\'/rootpath/frameworks\']', content)
         self.assertIn('"MyPkg_version" : "0.1"', content)

--- a/conans/test/unittests/client/generators/scons_test.py
+++ b/conans/test/unittests/client/generators/scons_test.py
@@ -45,7 +45,7 @@ class SConsGeneratorTest(unittest.TestCase):
         cpp_info.frameworks = ["cocoa"]
         cpp_info.frameworkdirs = ["frameworks"]
         cpp_info.filter_empty = False
-        conanfile.deps_cpp_info.add("MyPkg", cpp_info)
+        conanfile.deps_cpp_info.add("MyPkg", DepCppInfo(cpp_info))
         generator = SConsGenerator(conanfile)
         content = generator.content
         self.assertIn('"LIBS"        : [\'mypkg\', \'pthread\']', content)

